lib/generators/hyrax/templates/db/seeds.rb

```diff
@@ -9,89 +9,118 @@
-  parent.apply_depositor_metadata(user)
-end
-
-# Child Collection
-cc = Collection.create!(title: ["A Child Collection"], read_groups: ['public'], collection_type_gid: nestable_gid) do |child|
-  child.apply_depositor_metadata(user)
-  child.member_of_collections = [pc]
-end
-
-puts "Creating works"
-# Public
-3.times do |i|
-  GenericWork.create(title: ["Public #{i}"], read_groups: ['public']) do |work|
-    work.apply_depositor_metadata(user)
-    work.member_of_collections = [pnc]
-  end
-end
-
-# Authenticated
-2.times do |i|
-  GenericWork.create(title: ["Authenticated #{i}"], read_groups: ['registered']) do |work|
-    work.apply_depositor_metadata(user)
-    work.member_of_collections = [pc]
-  end
-end
-
-# Private
-1.times do |i|
-  GenericWork.create(title: ["Private #{i}"]) do |work|
-    work.apply_depositor_metadata(user)
-    work.member_of_collections = [cc]
-  end
-end
-
+# ---------------------------------
+# methods to create various objects
+# ---------------------------------
+def create_collection_type(machine_id, title, description, options)
+  coltype = Hyrax::CollectionType.find_by_machine_id(machine_id)
+  return coltype if coltype.present?
+  default_options = {
+      nestable: false, discoverable: false, sharable: false, allow_multiple_membership: false,
+      require_membership: false, assigns_workflow: false, assigns_visibility: false,
+      participants: [{ agent_type: Hyrax::CollectionTypeParticipant::GROUP_TYPE, agent_id: ::Ability.admin_group_name, access: Hyrax::CollectionTypeParticipant::MANAGE_ACCESS },
+                     { agent_type: Hyrax::CollectionTypeParticipant::GROUP_TYPE, agent_id: ::Ability.registered_group_name, access: Hyrax::CollectionTypeParticipant::CREATE_ACCESS }]
+  }
+  final_options = default_options.merge(options)
+  final_options['description'] = description
+  Hyrax::CollectionTypes::CreateService.create_collection_type(machine_id: machine_id, title: title, options: final_options)
+end
+
+def create_public_collection(user, type_gid, id, title, description, options={})
+  create_collection(user, type_gid, id, title, description, Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC, options)
+end
+
+def create_private_collection(user, type_gid, id, title, description, options={})
+  create_collection(user, type_gid, id, title, description, Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE, options)
+end
+
+def create_collection(user, type_gid, id, title, description, visibility, options={})
+  col = Collection.where(id: id)
+  return col.first if col.present?
+  col = Collection.new(id: id)
+  attrs = options
+  attrs[:title] = [title]
+  attrs[:description] = [description]
+  col.attributes = attrs
+  col.apply_depositor_metadata(user.user_key)
+  col.collection_type_gid = type_gid
+  col.visibility = visibility
+  col.save
+  Hyrax::Collections::PermissionsCreateService.create_default(collection: col, creating_user: user)
+  col.update_index
```

Comment:
Is there a reason this step isn’t a part of PermissionsCreateService? You always need to run it afterwards, right?